### ISSUE TITLE
When should `merge` be converted to `optionize`?

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/chipper#1226]
From https://github.com/phetsims/scenery/issues/1406#issuecomment-1099634315:

* When is it OK to continue to use `merge` vs converting to `optionize`?

I've encountered quite a few .ts files that are still using `merge`.

**Comment by samreid** [comment]
From the developer meeting notes:

> When is it OK to continue to use merge vs converting to optionize? (Tracking in [When should merge be converted to optionize? chipper#1226](https://github.com/phetsims/chipper/issues/1226).)

> MK: Good question.  I don’t know yet.  But merge is a tool that is not deprecated, at least not yet.
> In general, always use optionize when defining a class and setting option values.  In other cases, when manipulating options, optionize is still preferred due to the type safety it offers, but there may be some cases where merge is the only way to get things done.

See also related discussion in https://github.com/phetsims/phet-core/issues/114

**Comment by pixelzoom** [comment]
I agree that `merge` has uses, but should be avoided in relation to the "options" pattern. (I think that's what @zepumph is saying in the preceding comment.)  My TS code is `merge`-free in that regard, and I plan to keep it that way.

I'm not sure why this is being assigned back to me, because I don't see much in the way of consensus or next steps from the recent developer discussion.   So I'm going to unassign myself.

Remaining questions that occur to me:

* How to identify and address TS code where `merge` should changed to `optionize`?  Or does PhET want to bother?
* Tooling support (lint, etc.) to identify inappropriate uses of `merge`
* Anything relevant to add to the TS convensions doc?

**Comment by zepumph** [comment]
You were assigned back because you hosted the original question of differentiating when to use the two. I believe you were absent from the dev meeting when we discussed this so I wanted to make sure you were able to review @samreid's response. 

I updated convention doc and added a bad-text lint rule with 140 failures currently. This is largely lumped into the pot of many typescript lint rules with too many failures to turn on at the moment. At some point we are going to have to bite the bullet, but that is for a more general conversation with all devs over in https://github.com/phetsims/chipper/issues/1114. Closing
